### PR TITLE
Thomasht86/submit test zip

### PR DESCRIPTION
**Pull request by thomasht86**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

Submits applicationTestZip if `tests`-directory exists when calling `VespaCloud().deploy_to_prod()`.
The sample app used for integration test is duplicated from https://github.com/vespa-cloud/examples/tree/main/CI-CD/production-deployment-with-tests. 

Could skip the integration test (and duplicating the example) if it is considered superfluous to our other tests?

### CODE REVIEW

File: tests/testapps/production-deployment-with-tests/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<!-- Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root. -->
```
Comment (kkraune) [inline tests/testapps/production-deployment-with-tests/README.md:1]: copyright Vespa.ai

Comment (kkraune) [reply tests/testapps/production-deployment-with-tests/README.md:1]: in a few other files, too

**Review by kkraune — DISMISSED** [review]
👍

File: tests/testapps/production-deployment-with-tests/README.md (hunk 1 above)
Comment (thomasht86) [reply tests/testapps/production-deployment-with-tests/README.md]: nice catch! updated

**Review by kkraune — APPROVED** [review]
👍

**Pull request merged**